test/functional/rpc_getdescriptoractivity.py

```diff
@@ -0,0 +1,164 @@
+
+    def test_multiple_addresses(self, node, wallet):
+        _, spk_1, addr_1 = getnewdestination()
+        _, spk_2, addr_2 = getnewdestination()
+        wallet.send_to(from_node=node, scriptPubKey=spk_1, amount=1 * COIN)
+        wallet.send_to(from_node=node, scriptPubKey=spk_2, amount=2 * COIN)
+
+        blockhash = self.generate(node, 1)[0]
+
+        result = node.getdescriptoractivity([blockhash], [f"addr({addr_1})", f"addr({addr_2})"], True)
+
+        assert_equal(len(result['activity']), 2)
+
+        [a1] = [a for a in result['activity'] if a['address'] == addr_1]
+        [a2] = [a for a in result['activity'] if a['address'] == addr_2]
+
+        assert a1['blockhash'] == blockhash
+        assert a1['amount'] == 1.0
+
+        assert a2['blockhash'] == blockhash
+        assert a2['amount'] == 2.0
+
+    def test_invalid_blockhash(self, node, wallet):
+        self.generate(node, 20) # Generate to get more fees
+
+        _, spk_1, addr_1 = getnewdestination()
+        wallet.send_to(from_node=node, scriptPubKey=spk_1, amount=1 * COIN)
+
+        invalid_blockhash = "0000000000000000000000000000000000000000000000000000000000000000"
+
+        assert_raises_rpc_error(
+            -5, "Block not found",
+            node.getdescriptoractivity, [invalid_blockhash], [f"addr({addr_1})"], True)
+
+    def test_confirmed_and_unconfirmed(self, node, wallet):
+        self.generate(node, 20) # Generate to get more fees
+
+        _, spk_1, addr_1 = getnewdestination()
+        txid_1 = wallet.send_to(
+            from_node=node, scriptPubKey=spk_1, amount=1 * COIN)['txid']
+        blockhash = self.generate(node, 1)[0]
+
+        _, spk_2, to_addr = getnewdestination()
+        txid_2 = wallet.send_to(
+            from_node=node, scriptPubKey=spk_2, amount=1 * COIN)['txid']
+
+        result = node.getdescriptoractivity(
+            [blockhash], [f"addr({addr_1})", f"addr({to_addr})"], True)
+
+        activity = result['activity']
+        assert_equal(len(activity), 2)
+
+        [confirmed] = [a for a in activity if a['blockhash'] == blockhash]
+        assert confirmed['txid'] == txid_1
+        assert confirmed['height'] == node.getblockchaininfo()['blocks']
+
+        [unconfirmed] = [a for a in activity if not a['blockhash']]
+        assert unconfirmed['blockhash'] == ""
+        assert unconfirmed['height'] == -1
+
+        assert any(a['txid'] == txid_2 for a in activity if a['blockhash'] == "")
+
+    def test_receive_then_spend(self, node, wallet):
+        self.generate(node, 20) # Generate to get more fees
+
+        sent1 = wallet.send_self_transfer(from_node=node)
+        utxo = sent1['new_utxo']
+        blockhash_1 = self.generate(node, 1)[0]
+
+        sent2 = wallet.send_self_transfer(from_node=node, utxo_to_spend=utxo)
+        blockhash_2 = self.generate(node, 1)[0]
+
+        result = node.getdescriptoractivity(
+            [blockhash_1, blockhash_2], [wallet.get_descriptor()], True)
```

Comment:
block order matters, run this case twice, once with blocks in reverse history order?

Reply:
should also test that repeated blockhashes is acceptable(AFAICT it just repeats the event)

Reply:
Added tests including duplicate and reversal.